### PR TITLE
Add doccomments for cellInfo in grid cell templates

### PR DESCRIPTION
**Pull request by KiriosK**

### CODE REVIEW

File: js/ui/grid_core/ui.grid_core.columns_controller.js [hunk 1]
```diff
@@ -331,29 +331,112 @@ module.exports = {
...
+            * @type_function_param2_field1 data:object
+            * @type_function_param2_field2 component:dxTreeList
+            * @type_function_param2_field3 value:any
+            * @type_function_param2_field4 oldValue:any
+            * @type_function_param2_field5 displayValue:any
+            * @type_function_param2_field6 text:string
+            * @type_function_param2_field7 columnIndex:number
+            * @type_function_param2_field8 rowIndex:number
+            * @type_function_param2_field9 column:dxTreeListColumn
+            * @type_function_param2_field10 rowType:string
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.columns_controller.js]: Add doc comment for row field:  "row:dxTreeListRow"

File: js/ui/grid_core/ui.grid_core.columns_controller.js [hunk 2]
```diff
@@ -331,29 +331,112 @@ module.exports = {
...
+             * @type_function_param2_field1 setValue(newValue, newText):any
+             * @type_function_param2_field2 data:object
+             * @type_function_param2_field3 component:dxDataGrid
+             * @type_function_param2_field4 value:any
+             * @type_function_param2_field5 displayValue:any
+             * @type_function_param2_field6 text:string
+             * @type_function_param2_field7 columnIndex:number
+             * @type_function_param2_field8 rowIndex:number
+             * @type_function_param2_field9 column:dxDataGridColumn
+             * @type_function_param2_field10 rowType:string
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.columns_controller.js]: Add doc comment for row field:  "row:dxDataGridRow"
Add doc comment for watch field

File: js/ui/grid_core/ui.grid_core.columns_controller.js [hunk 3]
```diff
@@ -331,29 +331,112 @@ module.exports = {
...
+             * @type_function_param2_field1 setValue(newValue, newText):any
+             * @type_function_param2_field2 data:object
+             * @type_function_param2_field3 component:dxTreeList
+             * @type_function_param2_field4 value:any
+             * @type_function_param2_field5 displayValue:any
+             * @type_function_param2_field6 text:string
+             * @type_function_param2_field7 columnIndex:number
+             * @type_function_param2_field8 rowIndex:number
+             * @type_function_param2_field9 column:dxTreeListColumn
+             * @type_function_param2_field10 rowType:string
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.columns_controller.js]: Add doc comment for row field:  "row:dxTreeListRow"
Add doc comment for watch field

File: js/ui/grid_core/ui.grid_core.columns_controller.js [hunk 4]
```diff
@@ -331,29 +331,112 @@ module.exports = {
...
+             * @type_function_param2_field2 component:dxDataGrid
+             * @type_function_param2_field3 value:any
+             * @type_function_param2_field4 oldValue:any
+             * @type_function_param2_field5 displayValue:any
+             * @type_function_param2_field6 text:string
+             * @type_function_param2_field7 columnIndex:number
+             * @type_function_param2_field8 rowIndex:number
+             * @type_function_param2_field9 column:dxDataGridColumn
+             * @type_function_param2_field10 rowType:string
+             * @type_function_param2_field11 watch:function
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.columns_controller.js]: Add doc comment for row field: "row:dxDataGridRow"

File: js/ui/grid_core/ui.grid_core.columns_controller.js [hunk 5]
```diff
@@ -331,29 +331,112 @@ module.exports = {
...
              * @type_function_param1 cellElement:dxElement
              * @type_function_param2 cellInfo:object
+             * @type_function_param2_field1 component:dxDataGrid
+             * @type_function_param2_field2 value:any
+             * @type_function_param2_field3 text:string
+             * @type_function_param2_field4 displayValue:any
+             * @type_function_param2_field5 columnIndex:number
+             * @type_function_param2_field6 rowIndex:number
+             * @type_function_param2_field7 column:dxDataGridColumn
+             * @type_function_param2_field8 summaryItems:Array<any>
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.columns_controller.js]: Add doc comment for row field: "row:dxDataGridRow"

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/grid_core/ui.grid_core.columns_controller.js [hunk 6]
```diff
@@ -331,29 +331,119 @@ module.exports = {
...
+             * @type_function_param2_field11 rowType:string
+             * @type_function_param2_field12 watch:function
              */
+
+            /**
+             * @name dxTreeListColumn.editCellTemplate
+             * @type template|function
+             * @type_function_param1 cellElement:dxElement
+             * @type_function_param2 cellInfo:object
+             * @type_function_param2_field1 setValue(newValue, newText):any
```
Comment (viterobk) [inline js/ui/grid_core/ui.grid_core.columns_controller.js]: `@type_function_param2_field1 setValue:function`

File: js/ui/grid_core/ui.grid_core.columns_controller.js [hunk 7]
```diff
@@ -331,29 +331,119 @@ module.exports = {
...
+             * @type_function_param2_field3 column:dxTreeListColumn
              */
+
             /**
-             * @name GridBaseColumn.editCellTemplate
+             * @name dxDataGridColumn.editCellTemplate
              * @type template|function
              * @type_function_param1 cellElement:dxElement
              * @type_function_param2 cellInfo:object
+             * @type_function_param2_field1 setValue(newValue, newText):any
```
Comment (viterobk) [inline js/ui/grid_core/ui.grid_core.columns_controller.js]: `@type_function_param2_field1 setValue:function`
The function signature can be shown in the field's description.

**Review by viterobk — APPROVED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Review by vconst — APPROVED** [review]

**Pull request merged**